controller/transfer/association_transfer_manager.py

```diff
@@ -0,0 +1,111 @@
+from submodules.model import enums
+from submodules.model.business_objects import (
+    general,
+    labeling_task_label,
+    record_label_association,
+)
+from submodules.model.models import RecordLabelAssociation
+from controller.weak_supervision import weak_supervision_service as weak_supervision
+
+
+def import_associations(
+    project_id: str,
+    user_id: str,
+    model_name: str,
+    labeling_task_name: str,
+    associations: Dict[str, Any],
+    indices: Dict[str, Any],
+    source_type: str,
+) -> int:
+    labeling_task = labeling_task_manager.get_labeling_task_by_name(
+        project_id, labeling_task_name
+    )
+
+    information_source = information_source_manager.get_information_source_by_name(
+        project_id, model_name
+    )
+    if information_source is None:
+        if source_type == "model_callback":
+            information_source = information_source_manager.create_information_source(
+                project_id,
+                user_id,
+                labeling_task.id,
+                model_name,
+                "",
+                "This is a model callback",
+                enums.LabelSource.MODEL_CALLBACK.value,
+            )
+        elif source_type == "heuristic":
+            information_source = information_source_manager.create_information_source(
+                project_id,
+                user_id,
+                labeling_task.id,
+                model_name,
+                "",
+                "This is a heuristic",
+                enums.LabelSource.INFORMATION_SOURCE.value,
+            )
+        else:
+            raise Exception("Unknown source type")
+
+    attribute_names = list(indices[0].keys())
+    attribute_list = attribute_manager.get_all_attributes_by_names(
+        project_id, attribute_names
+    )
+
+    records = record_manager.get_records_by_composite_keys(
+        project_id,
+        indices,
+        attribute_list,
+        enums.RecordCategory.SCALE.value,  # we currently don't use TEST any more
+    )
+    record_ids = [record.id for record in list(records.values())]
+
+    record_label_associations = []
+    for record_id, association in zip(record_ids, associations):
+        label_name, confidence = association
+
+        label = labeling_task_label.get_by_name(
+            project_id, labeling_task.id, label_name
+        )
+
+        record_label_associations.append(
+            RecordLabelAssociation(
+                project_id=project_id,
+                record_id=record_id,
+                labeling_task_label_id=label.id,
+                source_type=enums.LabelSource.MODEL_CALLBACK.value,
+                source_id=information_source.id,
+                return_type=enums.InformationSourceReturnType.RETURN.value,
+                confidence=confidence,
+                created_by=user_id,
+            )
+        )
+
+    record_label_association.delete_by_source_id_and_record_ids(
+        project_id, information_source.id, record_ids
+    )
```

Comment:
is this wanted? only the provided record ids are removed so a newer import with only 10 record ids would still look like it has a lot of records corresponding to it?

Reply:
again, this is a result of the batching mechanism